hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/filedrops/RequestFileDropGenerater.java

```diff
@@ -22,16 +23,24 @@
 
 	@Handler
 	public void createFileDrops(Exchange exchange) {		
+		String methodName = LoggingUtil.getMethodName();
+		logger.debug("Beging {}", methodName);
+		
+		Object body = exchange.getIn().getBody();
 		/*
 		 * when we use wiretap then the tapped exchange has its own exchange id
 		 * But the wire tap will store the exchange id from its parent as a
 		 * "correlated exchange id".
 		 */
-		String corId = exchange.getProperty(Exchange.CORRELATION_ID, String.class);
-		String fileName = buildFileNameParameters(exchange,corId);
-		String requestFileName = fileName + REQUEST_FILE;
-		writeFiledrop(exchange.getIn().getBody().toString(), requestFileName);
-		logger.info("{} - TransactionId: {}, Successfully created file drops for request: {}",LoggingUtil.getMethodName(), exchange.getProperty(Exchange.CORRELATION_ID, String.class), requestFileName);
+		if (body != null && StringUtils.isNotBlank(body.toString())) {
```

Comment:
In situations like these I prefer to return early (for a negative condition) as it avoids having to nest code.

Reply:
Changes made as per comments